node/src/GlideClusterClient.ts

```diff
@@ -314,6 +318,151 @@ export class GlideClusterClient extends BaseClient {
         );
     }
 
+    /**
+     * @internal
+     */
+    protected scanOptionsToProto(
+        cursor: string,
+        options?: ScanOptions,
+    ): command_request.ClusterScan {
+        const command = command_request.ClusterScan.create();
+        command.cursor = cursor;
+
+        if (options?.match) {
+            command.matchPattern = Buffer.from(options.match);
+        }
+
+        if (options?.count) {
+            command.count = options.count;
+        }
+
+        if (options?.type) {
+            command.objectType = options.type;
+        }
+
+        return command;
+    }
+
+    /**
+     * @internal
+     */
+    protected createClusterScanPromise(
+        cursor: ClusterScanCursor,
+        options?: ScanOptions & DecoderOption,
+    ): Promise<[ClusterScanCursor, GlideString[]]> {
+        // separate decoder option from scan options
+        const { decoder, ...scanOptions } = options || {};
+        const stringDecoder: boolean =
+            (decoder ?? this.defaultDecoder) === Decoder.String;
+
+        if (this.isClosed) {
+            throw new ClosingError(
+                "Unable to execute requests; the client is closed. Please create a new client.",
+            );
+        }
+
+        const cursorId = cursor.getCursor();
+        const command = this.scanOptionsToProto(cursorId, scanOptions);
+        return new Promise((resolve, reject) => {
+            const callbackIndex = this.getCallbackIndex();
+            this.promiseCallbackFunctions[callbackIndex] = [
+                (resolveAns) => {
+                    try {
+                        resolveAns = this.getValueToResolve(
+                            resolveAns,
+                            stringDecoder,
+                        );
+                        resolveAns[0] = new ClusterScanCursor(
+                            resolveAns[0].toString(),
+                        );
+                        resolve(resolveAns);
+                    } catch (err) {
+                        Logger.log(
+                            "error",
+                            "Decoder",
```

Comment:
Are there any other errors possible here besides the Decoder error?

Reply:
ill break it to another try catch

Reply:
Ill try to think but i dont find any